File: JuansDevicesTests.py
#!/usr/bin/python
#2345678901234567890123456789012345678901234567890123456789012345678901234567890
#
# JuansDevicesTests.py - read a Juan Defined Device file.
#

Verbose = False

import os

import JuansDevices
 
testdevicelist = """
MT0	0,0,1,0
MT1	0,0,1,1
MT2	0,0,1,2
MT3	0,0,1,3
MT4	0,0,1,4
MT5	0,0,1,5
MT6	0,0,1,6
MT7	0,0,1,7
"""
testdevicearray = testdevicelist.split("\n")

#JuansDevices.JuansDevices.Verbose = True

import unittest

class JuansDevicesTests(unittest.TestCase):

	def test_Count_NewFromStringArray(self):
		dlo = JuansDevices.JuansDevices.NewFromStringArray(testdevicearray) 
		self.assertEqual(len(dlo.__dict__.keys()),8)

	def test_Values_NewFromStringArray(self):
		dlo = JuansDevices.JuansDevices.NewFromStringArray(testdevicearray) 
		self.assertEqual(dlo.MT0,"0,0,1,0")
		self.assertEqual(dlo.MT4,"0,0,1,4")
		self.assertEqual(dlo.MT7,"0,0,1,7")

	def test_Count_NewFromTSV_File(self):
		fspec = "%s/TestDevices.tsv" % os.environ['PWD']
		dlo = JuansDevices.JuansDevices.NewFromTSV_File(fspec)
		self.assertEqual(len(dlo.__dict__.keys()),100)

	def test_Value_NewFromTSV_File(self):
		fspec = "%s/TestDevices.tsv" % os.environ['PWD']
		dlo = JuansDevices.JuansDevices.NewFromTSV_File(fspec)
		self.assertEqual(dlo.MT7,"0,0,1,7")

	def test_Count_NewSubset(self):
		dlo = JuansDevices.JuansDevices.NewFromStringArray(testdevicearray) 
		dlo2 = dlo.NewSubset(("MT0","MT5"))
		self.assertEqual(len(dlo2.__dict__.keys()),2)

	def test_Value_NewSubset(self):
		dlo = JuansDevices.JuansDevices.NewFromStringArray(testdevicearray) 
		dlo2 = dlo.NewSubset(("MT0","MT5"))
		self.assertEqual(dlo2.MT0,"0,0,1,0")
		self.assertEqual(dlo2.MT5,"0,0,1,5")

	def test_Count_SubsetFromTSV_File(self):
		fspec1 = "%s/TestDevices.tsv" % os.environ['PWD']
		dlo = JuansDevices.JuansDevices.NewFromTSV_File(fspec1)
		fspec2 = "%s/TestDeviceSets.tsv" % os.environ['PWD']
		sslabel = "NuclearDevices"
		dlo2 = dlo.SubsetFromTSV_File(fspec2,sslabel)
		self.assertEqual(len(dlo2.__dict__.keys()),6)

	def test_Value_SubsetFromTSV_File(self):
		fspec1 = "%s/TestDevices.tsv" % os.environ['PWD']
		dlo = JuansDevices.JuansDevices.NewFromTSV_File(fspec1)
		fspec2 = "%s/TestDeviceSets.tsv" % os.environ['PWD']
		sslabel = "NuclearDevices"
		dlo2 = dlo.SubsetFromTSV_File(fspec2,sslabel)
		self.assertEqual(dlo2.MT9,"0,0,1,9")

def main():
    unittest.main()

if __name__ == "__main__":
    main()

#2345678901234567890123456789012345678901234567890123456789012345678901234567890
# End of JuansDevicesTests.py

File: JuansDevicesTests.pl
#!/usr/bin/perl
#2345678901234567890123456789012345678901234567890123456789012345678901234567890
#
# JuansDevicesTests.pl - read a Juan Defined Device file.
#

use strict;
use warnings;
use 5.016;

use Test::Simple tests => 9;
 
use JuansDevices;  # What you're testing.
 
# #### Test JuansDevices::NewFromStringArray:

my $sa = <<EOSA;
MT0	0,0,1,0
MT1	0,0,1,1
MT2	0,0,1,2
MT3	0,0,1,3
MT4	0,0,1,4
MT5	0,0,1,5
MT6	0,0,1,6
MT7	0,0,1,7
EOSA
my @sa = split("\n",$sa);

# #### Test JuansDevices::NewFromStringArray

my $dlo1 = JuansDevices::NewFromStringArray(@sa);

ok( $dlo1->{MT0} eq '0,0,1,0','NewFromStringArray(@sa) got first element of array.' );
ok( $dlo1->{MT4} eq '0,0,1,4','NewFromStringArray(@sa) got middle element of array.' );
ok( $dlo1->{MT7} eq '0,0,1,7','NewFromStringArray(@sa) got last element of array.' );

my $kc1 = keys %$dlo1;

ok( $kc1 == 8,'NewFromStringArray got 8 devices in the string array.' );

# #### Test JuansDevices::NewFromTSV_File

my $f1 = "$ENV{PWD}/TestDevices.tsv";
my $dlo2 = JuansDevices::NewFromTSV_File($f1);
ok( $dlo2->{MT7} eq '0,0,1,7',"NewFromTSV_FILE got line 8 from file $f1.");

my $kc2 = keys %$dlo2;
ok( $kc2 == 100,'NewFromTSV_FILE got 100 devices from the definition file.' );

# #### Test JuansDevices::NewHashSubset

my $dlo3 = JuansDevices::NewHashSubset($dlo1,("MT0","MT5"));
my $kc3 = keys %$dlo3;
ok( $kc3 == 2,'NewHashSubset() got 2 devices.' );

# #### Test JuansDevices::SubsetFromTSV_File

my $sslabel = "NuclearDevices";
my $dlo4 = $dlo2->SubsetFromTSV_File("$ENV{PWD}/TestDeviceSets.tsv",$sslabel);
my $kc4 = keys %$dlo4;
ok( $kc4 == 6,'SubsetFromTSV_File got 2 devices.' );
ok( $dlo2->{MT9} eq '0,0,1,9',"NewFromTSV_FILE got line 10 from file $f1.");

#
#2345678901234567890123456789012345678901234567890123456789012345678901234567890
# End of JuansDevicesTests.pl
